{
  "353": {
    "beta-0.1.096": {
      "linux": {
        "compilation_exit_code": 3,
        "compiler_error_output": "Error: Jai encountered an exception running the program and is terminating.\n\nThis happened outside compile-time execution, so it is likely the compiler's fault.\n\nStack trace of the compiler:\n\n\nStack trace:\n/home/ubuntu/Desktop/Jai/jai-beta-0.1.096/bin/jai-linux() [0x74510e]\n/lib/x86_64-linux-gnu/libc.so.6(+0x42520) [0x707597c42520]\n/home/ubuntu/Desktop/Jai/jai-beta-0.1.096/bin/jai-linux(_ZN5Typer11infer_typesEP15Ast_Declaration+0x4ca) [0x818aaa]\n/home/ubuntu/Desktop/Jai/jai-beta-0.1.096/bin/jai-linux(_ZN5Typer11infer_typesEP6Queued+0x1be) [0x80564e]\n/home/ubuntu/Desktop/Jai/jai-beta-0.1.096/bin/jai-linux() [0x7cd5e2]\n/home/ubuntu/Desktop/Jai/jai-beta-0.1.096/bin/jai-linux(_ZN9Workspace13do_pipe_inferEv+0xff) [0x7cd40f]\n/home/ubuntu/Desktop/Jai/jai-beta-0.1.096/bin/jai-linux(_ZN9Workspace10flow_pipesEb+0x11e) [0x7cfe3e]\n/home/ubuntu/Desktop/Jai/jai-beta-0.1.096/bin/jai-linux(_Z13compiler_pumpP9Workspaceb+0x38d) [0x7dc8ad]\n/home/ubuntu/Desktop/Jai/jai-beta-0.1.096/bin/jai-linux(_Z20do_compilation_cycleP9Workspaceb+0x52) [0x7defa2]\n/home/ubuntu/Desktop/Jai/jai-beta-0.1.096/bin/jai-linux(_Z28workspace_thread_entry_pointPv+0x1c0) [0x7df420]\n/home/ubuntu/Desktop/Jai/jai-beta-0.1.096/bin/jai-linux(_Z16thread_middlemanPv+0x35) [0x84a285]\n/lib/x86_64-linux-gnu/libc.so.6(+0x94ac3) [0x707597c94ac3]\n/lib/x86_64-linux-gnu/libc.so.6(+0x126850) [0x707597d26850]\n\n",
        "compiler_output": "",
        "expected_compilation_exit_code": 1,
        "expected_run_exit_code": -1,
        "is_runtime_test": false,
        "passed_test": false,
        "run_exit_code": -1,
        "runtime_error_output": "",
        "runtime_output": ""
      },
      "windows": {
        "compilation_exit_code": 3,
        "compiler_error_output": "Error: Jai encountered an exception running the program and is terminating.\r\n\r\nCommand line arguments:\r\n  C:/jai/bin/jai.exe compiler_bugs/353_352_CEC1.jai\r\n\nThis happened outside compile-time execution, so it is likely the compiler's fault.\n\nStack trace of the compiler:\n\n  unknown(C5E67374) : BaseThreadInitThunk\r\n\r\n",
        "compiler_output": "",
        "expected_compilation_exit_code": 1,
        "expected_run_exit_code": -1,
        "is_runtime_test": false,
        "passed_test": false,
        "run_exit_code": -1,
        "runtime_error_output": "",
        "runtime_output": ""
      }
    },
    "file_path": "compiler_bugs/353_352_CEC1.jai",
    "tracking_issue_number": 353
  },
  "356": {
    "beta-0.1.096": {
      "linux": {
        "compilation_exit_code": 1,
        "compiler_error_output": "\nIn Workspace 2 (\"Target Program\"):\nThe program contains circular dependencies.\nStrangely, we did not find a cycle, so this is probably due to a compiler bug. To help you work around the problem, we'll now print out what is not compiling:\n\nHere are items, waiting to compile, that look suspicious because we don't know why they are waiting:\n\n/home/ubuntu/Desktop/actions-runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/356_355_CEC0.jai:1,1: Info: (declaration of 'X')\n\n    X::union{x:X;}\n    ^^^^^^^^\n\nIn Workspace 2 (\"Target Program\"):\n/home/ubuntu/Desktop/actions-runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/356_355_CEC0.jai:1,1: s=0x0 type=DECLARATION\n\n    X::union{x:X;}\n    ^^^^^^^^\n\n/home/ubuntu/Desktop/actions-runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/356_355_CEC0.jai:1,10: Info: (declaration of 'x')\n\n    X::union{x:X;}\n             ^^^\n\nIn Workspace 2 (\"Target Program\"):\n/home/ubuntu/Desktop/actions-runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/356_355_CEC0.jai:1,10: s=0x0 type=DECLARATION\n\n    X::union{x:X;}\n             ^^^\n\n\nHere are some other items, that may just be waiting on the ones listed above:\n\n/home/ubuntu/Desktop/actions-runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/356_355_CEC0.jai:1,4: Info: (X)\n\n    X::union{x:X;}\n       ^^^^^\n\nIn Workspace 2 (\"Target Program\"):\n/home/ubuntu/Desktop/actions-runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/356_355_CEC0.jai:1,4: s=0x0 type=STRUCT\n\n    X::union{x:X;}\n       ^^^^^\n\n\n",
        "compiler_output": "",
        "expected_compilation_exit_code": 0,
        "expected_run_exit_code": -1,
        "is_runtime_test": false,
        "passed_test": false,
        "run_exit_code": -1,
        "runtime_error_output": "",
        "runtime_output": ""
      },
      "windows": {
        "compilation_exit_code": 1,
        "compiler_error_output": "\r\nIn Workspace 2 (\"Target Program\"):\r\nThe program contains circular dependencies.\r\nStrangely, we did not find a cycle, so this is probably due to a compiler bug. To help you work around the problem, we'll now print out what is not compiling:\r\n\r\nHere are items, waiting to compile, that look suspicious because we don't know why they are waiting:\r\n\r\nC:/Users/Private/Desktop/testo_runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/356_355_CEC0.jai:1,1: Info: (declaration of 'X')\r\n\r\n    X::union{x:X;}\r\n    ^^^^^^^^\r\n\r\nIn Workspace 2 (\"Target Program\"):\r\nC:/Users/Private/Desktop/testo_runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/356_355_CEC0.jai:1,1: s=0x0 type=DECLARATION\r\n\r\n    X::union{x:X;}\r\n    ^^^^^^^^\r\n\r\nC:/Users/Private/Desktop/testo_runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/356_355_CEC0.jai:1,10: Info: (declaration of 'x')\r\n\r\n    X::union{x:X;}\r\n             ^^^\r\n\r\nIn Workspace 2 (\"Target Program\"):\r\nC:/Users/Private/Desktop/testo_runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/356_355_CEC0.jai:1,10: s=0x0 type=DECLARATION\r\n\r\n    X::union{x:X;}\r\n             ^^^\r\n\r\n\r\nHere are some other items, that may just be waiting on the ones listed above:\r\n\r\nC:/Users/Private/Desktop/testo_runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/356_355_CEC0.jai:1,4: Info: (X)\r\n\r\n    X::union{x:X;}\r\n       ^^^^^\r\n\r\nIn Workspace 2 (\"Target Program\"):\r\nC:/Users/Private/Desktop/testo_runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/356_355_CEC0.jai:1,4: s=0x0 type=STRUCT\r\n\r\n    X::union{x:X;}\r\n       ^^^^^\r\n\r\n\r\n",
        "compiler_output": "",
        "expected_compilation_exit_code": 0,
        "expected_run_exit_code": -1,
        "is_runtime_test": false,
        "passed_test": false,
        "run_exit_code": -1,
        "runtime_error_output": "",
        "runtime_output": ""
      }
    },
    "file_path": "compiler_bugs/356_355_CEC0.jai",
    "tracking_issue_number": 356
  },
  "365": {
    "beta-0.1.096": {
      "linux": {
        "compilation_exit_code": 1,
        "compiler_error_output": "\nIn Workspace 2 (\"Target Program\"):\n/home/ubuntu/Desktop/actions-runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/365_363_CEC0.jai:3,33: Error: Compile-time assertion failed.\n\n    #import \"Basic\";\n    check :: (o1 : $T1, o2 : $T2) { #assert(T1 == T2); }\n                                    ^^^^^^^\n\nInfo: In Workspace 2 (\"Target Program\"), building target '365_363_CEC0'.\n/home/ubuntu/Desktop/actions-runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/365_363_CEC0.jai:3,10: Polymorph generation history:\n/home/ubuntu/Desktop/actions-runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/365_363_CEC0.jai:3,10: Info: While generating a polymorph of this procedure.\n\n    check :: (o1 : $T1, o2 : $T2) { #assert(T1 == T2); }\n             ^^^^^^^^^^^^^^^^^^^^\n\n/home/ubuntu/Desktop/actions-runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/365_363_CEC0.jai:3,10: ... with the following constants: (T1=void, T2=Type).\n/home/ubuntu/Desktop/actions-runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/365_363_CEC0.jai:6,2: Info: ... The polymorph was triggered here.\n\n    main :: () {\n    \tcheck((cast(*void)0).*, #run (cast(*void)0).*); // OK\n    \tcheck((struct{x:void;}).{}.x, #run (struct{x:void;}).{}.x); // fails\n     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n\n",
        "compiler_output": "",
        "expected_compilation_exit_code": 0,
        "expected_run_exit_code": -1,
        "is_runtime_test": false,
        "passed_test": false,
        "run_exit_code": -1,
        "runtime_error_output": "",
        "runtime_output": ""
      },
      "windows": {
        "compilation_exit_code": 1,
        "compiler_error_output": "\r\nIn Workspace 2 (\"Target Program\"):\r\nC:/Users/Private/Desktop/testo_runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/365_363_CEC0.jai:3,33: Error: Compile-time assertion failed.\r\n\r\n    #import \"Basic\";\r\n    check :: (o1 : $T1, o2 : $T2) { #assert(T1 == T2); }\r\n                                    ^^^^^^^\r\n\r\nInfo: In Workspace 2 (\"Target Program\"), building target '365_363_CEC0'.\r\nC:/Users/Private/Desktop/testo_runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/365_363_CEC0.jai:3,10: Polymorph generation history:\r\nC:/Users/Private/Desktop/testo_runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/365_363_CEC0.jai:3,10: Info: While generating a polymorph of this procedure.\r\n\r\n    check :: (o1 : $T1, o2 : $T2) { #assert(T1 == T2); }\r\n             ^^^^^^^^^^^^^^^^^^^^\r\n\r\nC:/Users/Private/Desktop/testo_runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/365_363_CEC0.jai:3,10: ... with the following constants: (T1=void, T2=Type).\r\nC:/Users/Private/Desktop/testo_runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/365_363_CEC0.jai:6,2: Info: ... The polymorph was triggered here.\r\n\r\n    main :: () {\r\n    \tcheck((cast(*void)0).*, #run (cast(*void)0).*); // OK\r\n    \tcheck((struct{x:void;}).{}.x, #run (struct{x:void;}).{}.x); // fails\r\n     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\r\n\r\n",
        "compiler_output": "",
        "expected_compilation_exit_code": 0,
        "expected_run_exit_code": -1,
        "is_runtime_test": false,
        "passed_test": false,
        "run_exit_code": -1,
        "runtime_error_output": "",
        "runtime_output": ""
      }
    },
    "file_path": "compiler_bugs/365_363_CEC0.jai",
    "tracking_issue_number": 365
  }
}